### PR TITLE
feat(yarn): yarn support for workspaces

### PR DESCRIPTION
**Pull request by easadev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/nativescript-cli/blob/master/CONTRIBUTING.md#contribute-to-the-code-base
- [ ] Tests for the changes are included.

## What is the new behavior?

Fixes/Implements/Closes #[Issue Number].

### CODE REVIEW

**Comment by easadev** [comment]
PR for this https://github.com/NativeScript/nativescript-cli/issues/4667

**Comment by easadev** [comment]
Seems like I need a git course. I tried amending the commit, but it resulted in a brand new commit, and I still cant get the old commit verifed.. should I delete this PR? or is there a way to fix it. Thank you and sorry for the noise

**Comment by rosen-vladimirov** [comment]
test cli-smoke

**Comment by rosen-vladimirov** [comment]
cla-bot check

**Comment by rosen-vladimirov** [comment]
@cla-bot check

File: lib/services/plugins-service.ts [hunk 1]
```diff
@@ -228,7 +228,10 @@ export class PluginsService implements IPluginsService {
 	}
 
 	private getPackageJsonFilePathForModule(moduleName: string, projectDir: string): string {
-		return path.join(this.getNodeModulesPath(projectDir), moduleName, "package.json");
+		const pathToJsonFile = require.resolve(`${moduleName}/package.json`, {
+				paths: [projectDir]
```
Comment (rosen-vladimirov) [inline lib/services/plugins-service.ts]: note: this code will work with Node.js 8.9 or later.

**Comment by rosen-vladimirov** [comment]
test cli-plugin cli-templates cli-preview

**Comment by rosen-vladimirov** [comment]
@easadev thanks a lot for your contribution. We'll run some tests and merge your PR. Meanwhile, it looks like one of your commits is with incorrect username, can you please squash all commits and push-force the branch, so we can ensure the CLA is signed.

**Comment by esanchez-endava** [comment]
Hi, @rosen-vladimirov Thank you for reviewing this so quickly. I just squashed the commits and force pushed them as you suggested.

**Comment by easadev** [comment]
Hi, @rosen-vladimirov Thank you for reviewing this so quickly. I just squashed the commits and force pushed them as you suggested.

**Comment by rosen-vladimirov** [comment]
test cli-plugin cli-templates cli-preview

**Comment by easadev** [comment]
Hi @rosen-vladimirov I noticed that one check failed (ci/jenkins/nativescript-cli) but I cant see the details. Can I know what it was so that I can fix it? Thank you

**Comment by rosen-vladimirov** [comment]
test cli-plugin

**Comment by rosen-vladimirov** [comment]
Hey @easadev ,
Thank you very much for your contribution and please excuse me for the delayed reply. I've rerun the tests and they pass, so I've merged the PR. The feature will be included in the next official release (6.0.0).
Thanks a lot!

**Pull request merged**